packages/garbo/src/fights.ts

```diff
@@ -533,7 +535,9 @@ export function dailyFights(): void {
         // use obtuse angel if have + have quake of arrows, otherwise reanimator
         // quake of arrows is PvP-stealable and costs ~50k, so don't assume we have it
         let bestCopier: Familiar | undefined;
-        if (
+        if (ChestMimic.have() && shouldMakeEgg(true)) {
+          bestCopier = $familiar`Chest Mimic`;
+        } else if (
```

Comment:
I'm not sure how to do this. I tried the below:

`function bestCopier(): Familiar | undefined {

  let bestCopier: Familiar | undefined;

  const location = new EmbezzlerFightRunOptions(nextFight).location;
  const underwater = location.environment === "underwater";
  const shouldCopy = get("_badlyRomanticArrows") === 0 && !underwater;

  // use obtuse angel if have + have quake of arrows, otherwise reanimator
  // quake of arrows is PvP-stealable and costs ~50k, so don't assume we have it

  if(ChestMimic.have() && $familiar`Chest Mimic`.experience >= 50 && get("_mimicEggsObtained") < 11) {
    bestCopier = $familiar`Chest Mimic`;
  } else if ( have($familiar`Obtuse Angel`) &&
          (familiarEquippedEquipment($familiar`Obtuse Angel`) ===
            $item`quake of arrows` ||
            retrieveItem($item`quake of arrows`)) && shouldCopy
        ) {
          bestCopier = $familiar`Obtuse Angel`;
        } else if (have($familiar`Reanimated Reanimator`) && shouldCopy) {
          bestCopier = $familiar`Reanimated Reanimator`;
        }
  return bestCopier;

}`

but this is very clearly not right, and on several levels!

Reply:
When you slice something out into a function you can use early returns to avoid using `let` variables:
```typescript
function familiarSpec(underwater: boolean): OutfitSpec {
  if (!underwater) {
    if (ChestMimic.have() && $familiar`Chest Mimic`.experience >= 50 && get("_mimicEggsObtained") < 11) {
    return { familiar: $familiar`Chest Mimic` };
    }
    if (get("_badlyRomanticArrows") === 0) {
      if (have($familiar`Obtuse Angel` && (familiarEquippedEquipment($familiar`Obtuse Angel`) === $item`quake of arrows` || retrieveItem($item`quake of arrows`)) {
        return { familiar: $familiar`Obtuse Angel`, famequip: $item`quake of arrows` };
      }
      if (have($familiar`Reanimated Reanimator`)) return { familiar: $familiar`reanimated reanimator` }
    }
  }
  return { familiar: meatFamiliar() }
}
```

Reply:
As a side note, was 50 a typo here or is that the plan?

Reply:
That's the plan. If we're fighting a KGE and we have 50 experience, we can make an egg, so let's make an egg.